text/0073-keep-key-logs.md

```diff
@@ -0,0 +1,63 @@
+# RFC: Keep Key Logs
+
+- RFC PR: 
+- Tracking issue:
+
+## Motivation
+
+To address https://github.com/tikv/tikv/issues/10841, we decide to prioritize availability over log completeness thus switch the overflow strategy of logger from `Block` to `Drop`. 
+
+However, some logs are useful for diagnosis in practice. We heavily depend on logs to investigate bugs or abnormalities of the system. If all logs are dropped randomly, reasoning about the raftstore or transaction issues may be impossible.
+
+Therefore, it is only a temporary workaround to simply drop overflowed logs. We hope to keep logs that are important for diagnosis if possible.
+
+## Detailed design
+
+The problem is that we don't define which logs are important. Whether a log is important for diagnosis is not related to the log level. 
+
+For example, we prefer to keep info level logs in the raftstore rather than keep warn level logs in the coprocessor module. The former one is critical for finding out how each region evolves while the latter one may be only a retryable error.
+
+So, these important logs deserve to have another set of macros:
+
+- `key_crit!`
+- `key_error!`
+- `key_warn!`
+- `key_info!`
+- `key_debug!`
+- `key_trace!`
+
+Logs from these macros should not be dropped as they are key information. These macros should **NOT** be abused.
```

Comment:
I am concerned about this design.

IMHO, the standard logging levels from `TRACE` to `FATAL` is so popular in the whole industry, that it's a bad idea to reinvent one.

For example ,when logging a message in `INFO` level, it means that we want people seeing this log message to know that the system is in certain state that the message says, like "I am ready to begin a transaction". Other verbose messages which is unnecessary in this level is logged into `DEBUG` or `TRACE`.

If there's an "important"(from this RFC's context) message that's important for diagnosis, we should considering put it into `WARN` which signals some unexpected things are happening, actions must be taken, but no direct damage. Or we can put it into `INFO` like the above, because it's important to say that the system is currently in some state for diagnosis reason.

I believe the problem then is that there are so much `INFO` level log messages, which overwhelms the important diagnosis message. But wait! Why don't we take a look at all those `INFO` level log messages, are they too verbose? I mean, do we really need to talk to the user all the time when we are doing some common operations?

PS: Those are my personal comments, which maybe wrong, please point out. The whole point is that it's really weird to reinvent a whole bunch of logging levels XD